spec/formatting.md

```diff
@@ -29,15 +129,114 @@ the local variable takes precedence.
 It is an error for a local variable definition to
 refer to a local variable that's defined after it in the message.
 
+The resolution of a _variable_ MAY fail if no value is identified for its _name_.
+If this happens, an Unresolved Variable error MUST be emitted.
+If a _variable_ would resolve to a _fallback value_,
+this MUST also be considered a failure.
+
+### Function Resolution
+
+To resolve the value of an _expression_ with a _function_ _annotation_,
```

Comment:
```suggestion
To resolve an _expression_ with a _function_ _annotation_,
```
As I understand the earlier definition, values aren't resolved; expressions are resolved to values.